src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UnpatchedVulnerabilities.java

```diff
@@ -44,6 +60,71 @@ protected Feature supportedFeature() {
   @Override
   protected Value fetchValueFor(GitHubProject project) {
     logger.info("Figuring out if the project has any unpatched vulnerability ...");
-    return UNPATCHED_VULNERABILITIES.value(unpatchedVulnerabilities.getFor(project.url()));
+
+    Vulnerabilities vulnerabilities = new Vulnerabilities();
+    vulnerabilities.add(knownUnpatchedVulnerabilities.getFor(project.url()));
+    vulnerabilities.add(vulnerabilitiesFromNvdFor(project));
+
+    return UNPATCHED_VULNERABILITIES.value(vulnerabilities);
   }
+
+  /**
+   * Looks for unpatched vulnerabilities in NVD for a specified project.
+   *
+   * @param project The project.
+   * @return Unpatched vulnerabilities.
+   */
+  Vulnerabilities vulnerabilitiesFromNvdFor(GitHubProject project) {
+    Vulnerabilities vulnerabilities = new Vulnerabilities();
+    for (NvdEntry entry : nvd.search(VendorDataMatcher.with(project))) {
+      if (isUnpatched(entry)) {
+        vulnerabilities.add(
```

Comment:
At this point, logically speaking they should also be added into knownUnpatchedVulnerabilities right? So that they will be taken from the same for the next search? Also shouldn't we filter those vulnerabilities which are already part of knownUnpatchedVulnerabilities?

Reply:
> At this point, logically speaking they should also be added into knownUnpatchedVulnerabilities right? So that they will be taken from the same for the next search? 

No, `knownUnpatchedVulnerabilities` reads vulnerabilities from a file. The file is updated manually.

> Also shouldn't we filter those vulnerabilities which are already part of knownUnpatchedVulnerabilities?

No, the `Vulnerabilities` class merges vulnerabilities by IDs (in the `fetchValueFor()` method).